src/OrchardCore/OrchardCore.Abstractions/Pooling/StringBuilderPool.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Diagnostics;
+using System.Text;
+
+namespace OrchardCore.Abstractions.Pooling
+{
+    /// <summary>
+    /// The usage is:
+    ///        var inst = PooledStringBuilder.GetInstance();
+    ///        var sb = inst.builder;
+    ///        ... Do Stuff...
+    ///        ... sb.ToString() ...
+    ///        inst.Free();
+    /// </summary>
+    internal sealed class StringBuilderPool : IDisposable
+    {
+        private const int DefaultCapacity = 1 * 1024;
+
+        // global pool
+        private static readonly ObjectPool<StringBuilderPool> s_poolInstance = CreatePool();
+
+        public readonly StringBuilder Builder = new StringBuilder(DefaultCapacity);
+        private readonly ObjectPool<StringBuilderPool> _pool;
+
+        private StringBuilderPool(ObjectPool<StringBuilderPool> pool)
+        {
+            Debug.Assert(pool != null);
+            _pool = pool;
+        }
+
+        public int Length => Builder.Length;
+
+        internal static ObjectPool<StringBuilderPool> CreatePool(int size = 1000)
```

Comment:
Maybe taken from another context, but 1000 is too big for this one. There is a single content item deserialized/serialized at a time per request. So unless we expect 1000 concurrent requests ... RPS doesn't got up after 8 right now. Si I would say 32 max. Need to check the size of the buffer too for content items. I believe the agency's home page should be bigger than 1KB.

Reply:
Agency homepage is 10KB, so this pool would not help.

Reply:
Updated to 32/32KB to run the benchmarks

Reply:
This is with a single connection before/after
```
| Max GC Heap Size (MB)         | 303           |    ▁▁▁▁▃▁▂▃▂▃▄▅▅▆▇█▂▃▄▄▅▆▂▂▃▂▂▂▂▂▁▁▁▁▃▂▂▂ |
| Max Number of Gen 0 GCs / sec | 4.00          |     ▂▄ ▂█  ▂       ▂     ▄  ▂▂▂▂▂▂▂▂▂ ▂   |
| Max Number of Gen 1 GCs / sec | 1.00          |      █  █  █             █  █             |
| Max Number of Gen 2 GCs / sec | 1.00          |         █                █                |
| Max Time in GC (%)            | 3.00          |      ▃▃ ███                               |
| Max Gen 0 Size (B)            | 9,735,824     |                                 █         |
| Max Gen 1 Size (B)            | 17,358,440    |     ▃  ▄▃▃▃█████████████████              |
| Max Gen 2 Size (B)            | 35,886,016    |      ▂▂▂▃▃▃▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄█████████████ |
| Max LOH Size (B)              | 614,392       |     ▃▃▃██████████████████████████████████ |
| Max Allocation Rate (B/sec)   | 299,604,616   |    ▁▂▄ ▃█▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▂▂▂▂▂▂▂▂▂▁   |

| Max GC Heap Size (MB)         | 247           |   ▁▁▁▁▁▂▃▂▃▄▅▆▇█▂▃▄▂▃▁▂▃▁▂▃▂▂▂▁▄▄▃▃▃▃▃▃▃ |
| Max Number of Gen 0 GCs / sec | 3.00          |    ▅▃ █▅ ▃      ▃  ▃ ▃  ▃  ▃▃▃▃ ▃▃▃▃▃    |
| Max Number of Gen 1 GCs / sec | 1.00          |    █  █  █         █                     |
| Max Number of Gen 2 GCs / sec | 1.00          |       █            █                     |
| Max Time in GC (%)            | 9.00          |    ▄  █▅▅                                |
| Max Gen 0 Size (B)            | 288           |    █████████████████████████████████████ |
| Max Gen 1 Size (B)            | 13,694,688    |        ▆▆▇▇▇▇▇▇▇███               ▁▁▁▁▁▁ |
| Max Gen 2 Size (B)            | 35,550,184    |    ▂▂▂▃▃▃▅▅▅▅▅▅▅▅▅▅█████████████████████ |
| Max LOH Size (B)              | 614,392       |    ▃▃▃██████████████████████████████████ |
| Max Allocation Rate (B/sec)   | 229,831,464   |   ▁▅▃ █▇▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▂▃▃▃▃▃▃▃▃▃▁   |
```

Reply:
I was much lower originally https://github.com/OrchardCMS/OrchardCore/commit/53d3e22ff4a0d404ba7170e4f3815761e8594ebd

Reply:
I think I'll do a commit for switching to zstring and we'll see how it behaves